downstairs/src/extent_inner_sqlite.rs

```diff
@@ -1565,8 +1574,11 @@ mod test {
             };
             let resp = inner.read(JobId(21), read, IOV_MAX_TEST)?;
 
-            // We should not get back our data, because block 0 was written.
-            assert_ne!(resp.blocks, vec![Some(block_context)]);
+            // We should get back our old data, because block 0 was written.
+            assert_eq!(
+                resp.blocks,
+                vec![ReadBlockContext::Unencrypted { hash: prev_hash }]
```

Comment:
It looks like the value being tested changed here - was this test wrong before? How did this previously pass?

Reply:
Previously, it was
```rust
assert_ne!(resp.blocks, vec![Some(block_context)]);
```
i.e. checking that we **don't** read back the new value

Now, it's `assert_eq!` and checks that we **do** read back the old value.

Reply:
Cool cool